### PR TITLE
fix(labs/ssr): Include template and doc reference in partIndex errors

### LINKED ISSUE #4012: [labs/ssr] Better errors on partIndex mismatch
### Should this be an RFC?

- [X] This is not a substantial change

### Which package is this a feature request for?

SSR (@lit-labs/ssr)

### Description

When hitting errors as reported in https://github.com/lit/lit/issues/3654 or https://github.com/lit/lit/issues/2441, it's hard to pinpoint where this happened just from the stack trace.

It would be good to also log the template to help locate what's causing this.

### Alternatives and Workarounds

Trial and error by rendering bits of template at a time.

### PR DESCRIPTION
**Pull request by jimsimon**
Fixes #4012

This PR adds the template causing the problem as well as some additional information about why the error might be occurring.

### CODE REVIEW

File: packages/labs/ssr/src/lib/render-value.ts [hunk 1]
```diff
@@ -759,10 +759,30 @@ function* renderTemplateResult(
...
+  }
+}
+
+function throwErrorForPartIndexMismatch(
+  partIndex: number,
+  result: TemplateResult
+) {
+  let template = result.strings[0];
+  for (let i = 0; i < result.values.length; i++) {
+    template += result.values[i] + result.strings[i + 1];
```
[the remark this answers is not in the document]
Comment (jimsimon) [reply packages/labs/ssr/src/lib/render-value.ts]: I went back and forth on whether to use the actual values or not. I think your reasoning for not doing so makes sense. It also makes the output a little more terse.

**Review by AndrewJakubowicz — APPROVED** [review]
Nice!

**Comment by jimsimon** [comment]
> Nice!
> 
> Would you be able to add a changeset entry?

Just did! Hopefully I did it right. :)

**Review by AndrewJakubowicz — APPROVED** [review]
Nice! Thank you so much!

**Review by rictic — APPROVED** [review]

**Pull request merged**